api4/user.go

```diff
@@ -1720,6 +1730,35 @@ func login(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(user.ToJson()))
 }
 
+func extractLoginParameters(r *http.Request) map[string]string {
+	parameters := map[string]string{}
+	r.ParseForm()
+	if len(r.Form) > 0 {
+		for key, value := range r.Form {
+			if key == "login_id" {
+				parameters["login_id"] = value[0]
+			}
+			if key == "cws_token" {
+				parameters["cws_token"] = value[0]
+			}
+		}
+	} else {
+		props := model.MapFromJson(r.Body)
+		parameters["id"] = props["id"]
+		parameters["login_id"] = props["login_id"]
+		parameters["password"] = props["password"]
+		parameters["token"] = props["token"]
+		parameters["device_id"] = props["device_id"]
+		parameters["ldap_only"] = props["ldap_only"]
+	}
+
+	return parameters
+}
+
+func isCWSLogin(c *Context, cwsToken string) bool {
+	return *c.App.Config().ServiceSettings.EnableCWSTokenLogin && cwsToken != ""
```

Comment:
We should also check that the license feature `Cloud` is `true` too. Actually we could just check the license flag and not use a config setting at all, what do you think?

Reply:
If we're going to have this feature available for all the cloud installations would be better to use it yes

Reply:
All installations will need it so yeah let's do that then

Reply:
That would also prevent it being used on-prem